cbr/cbr.go

```diff
@@ -287,3 +297,66 @@ func (storageCBR *StorageCBR) DeleteCBRRuleZone(ruleID string, zoneID string) er
 
 	return err
 }
+
+// DeleteCBRZoneWithPattern ...
+func (storageCBR *StorageCBR) DeleteCBRZoneWithPattern(zoneDescription string) ([]string, error) {
+	//List all CBR zones
+	var zoneIds []string
+	listOptions := storageCBR.contextBasedRestrictionsService.NewListZonesOptions(storageCBR.accountID)
+	listzone, _, err := storageCBR.contextBasedRestrictionsService.ListZones(listOptions)
+	if err != nil {
+		storageCBR.logger.Error("Error while listing the zones", zap.Error(err))
+		return []string{}, err
+	}
+
+	storageCBR.logger.Info("total zone count", zap.Any("count", *listzone.Count))
+	if zoneDescription == "" {
```

Comment:
@skorati lets not consider empty description as delete all flag. As of now keep it pattern matching and lets skip the deletion of user provides no description.